0016/README.md

```diff
@@ -0,0 +1,488 @@
+        },
+        {
+            "name":"username",
+            "type":[
+                "string",
+                "null"
+            ],
+            "doc":"Client's username for validation"
+        },
+        {
+            "name":"password",
+            "type":[
+                "string",
+                "null"
+            ],
+            "doc":"Client's password for validation"
+        }
+    ]
+}
+```
+
+
+#### Client username and password validation response
+
+*Client username and password validation response* message MUST be sent by provider in response to an [client username and password validation request message](#client-username-and-password-validation-request).
+Provider MUST publish client username and password validation response message to the subject provided in the NATS `replyTo` field of the request.
+
+Client username and password validation response message payload MUST be an Avro-encoded object with the following schema ([0016-client-username-password-validation-response.avsc](./0016-client-username-password-validation-response.avsc)):
+```json
+{
+    "namespace":"org.kaaproject.ipc.ecap.gen.v1",
+    "name":"ClientUsernamePasswordValidationResponse",
+    "type":"record",
+    "doc":"Client username/password combination validation response message",
+    "fields":[
+        {
+            "name":"correlationId",
+            "type":"string",
+            "doc":"Message ID primarily used to track message processing across services"
+        },
+        {
+            "name":"timestamp",
+            "type":"long",
+            "doc":"Message creation UNIX timestamp in milliseconds"
+        },
+        {
+            "name":"timeout",
+            "type":"long",
+            "default":0,
+            "doc":"Amount of milliseconds (since the timestamp) until the message expires. Value of 0 is reserved to indicate no expiration."
+        },
+        {
+            "name":"clientId",
+            "type":[
+                "string",
+                "null"
+            ],
+            "doc":"ID of the client resolved by provided username/password combination. May be null in case the combination is not found."
+        },
+        {
+            "name":"statusCode",
+            "type":"int",
+            "doc":"HTTP status code of the request processing"
+        },
+        {
+            "name":"reasonPhrase",
+            "type":[
+              "null",
+              "string"
+            ],
+            "default":null,
+            "doc":"Human-readable status reason phrase"
+        }
+    ]
+}
+```
+
+
+### Client authentication by SSL certificate
+
+#### Client certificate validation request
+
+*Client certificate validation request* message is a targeted message that consumer sends to provider to make sure that the client is allowed to connect and to resolve the client ID.
```

Comment:
Add a note that the consumer must already check the certificate validity?

Reply:
Good idea.